### PR TITLE
My Jetpack: handle redirect when no connection

### PR DESCRIPTION
**Pull request by retrofox**
# ~Follow up of https://github.com/Automattic/jetpack/pull/22530~

This PR adds changes about how the app handles the redirect when it doesn't have a site and a user connected.
Relevant points:

* It adds an optional `redirect` option to the `useMyJetpackConnection()` hook. When it's true, the client will perform a redirect to the Jetpack dashboard in order to offer the user to connect to their site.
* it checks at top of the app whether there aren't site and user connected. If so, it won't render anything and wait for the client redirect.
* Because of the above, I've removed the redirecting handling from the `<ConnectionsSection />` component.

Fixes #

#### Changes proposed in this Pull Request:

* My Jetpack: handle redirect when no connection

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Check connecting and disconnecting the site.
* Confirm when it's disconnected, the app performs a redirect to the Jetpack dashboard
* In short, disconnect, connect, etc should work as expected

### CODE REVIEW

**Review by leogermani — APPROVED** [review]
Tests well in all situations

* fully connected
* connected at a site level
* redirect after disconnect

**Review by leogermani — APPROVED** [review]

**Review by leogermani — APPROVED** [review]
